### PR TITLE
Snapshot testing on CI

### PR DESCRIPTION
**Pull request by gert-janvercauteren**
# Introduce Snapshot testing on CI

When we add `take-snapshot` label to the PR we will retake snapshots to facilitate easier PR management

How this works:
- Make a commit with 'Record snapshots'
- it will then remove all __Snapshot__/*.png files
- re-run the tests
- 1 commit will show up when snapshots changed + a comment to make you aware
- Then close/re-open the pr to get everything running and all should be peachy.

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: .github/workflows/_build.yml [hunk 1]
```diff
@@ -11,9 +11,6 @@ jobs:
   Setup:
     name: Install Dependencies
     runs-on: macos-latest
-    permissions:
-      statuses: write
-      pull-requests: write
```
Comment (gert-janvercauteren) [inline .github/workflows/_build.yml:16]: These are already passed down from pr/main workflow

File: .github/workflows/_build.yml [hunk 2]
```diff
@@ -135,45 +132,8 @@ jobs:
...
-      - name: Pod Install
-        run: bundle exec pod install
-        working-directory: Example
-
-      - name: Select Xcode 14
-        run: sudo xcode-select --switch /Applications/Xcode_14.2.app
-
-      - name: Run Tests
-        run: ./scripts/ci ${{ matrix.schemes }} test
+    uses: ./.github/workflows/_test.yml
```
Comment (gert-janvercauteren) [inline .github/workflows/_build.yml:136]: Introduce reusable flow

File: .github/workflows/_test.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        if: ${{inputs.retake_snapshots}}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git remote set-url origin https://x-access-token:$GITHUB_TOKEN@github.com/${{ github.repository }}
+          git fetch origin $GITHUB_HEAD_REF
+          git checkout $GITHUB_HEAD_REF
+          git add --ignore-removal */**/__Snapshots__/*.png
```
Comment (gert-janvercauteren) [inline .github/workflows/_test.yml:75]: Only add changed files, skip deleted.

Comment (marianeum) [reply .github/workflows/_test.yml:75]: what if somebody deletes a snapshot test (ie for a deleted component or if we rename a test or similar)?

Comment (frugoman) [reply .github/workflows/_test.yml:74]: Good point ☝️

Comment (gert-janvercauteren) [reply .github/workflows/_test.yml]: This is no longer an issue. I have reverted back to also commiting deleted files.

**Comment by frugoman** [comment]
Record snapshots

**Review by frugoman — APPROVED** [review]
Looking good!! Lets get it merged and tweak as needed iteratively!

File: .github/workflows/_test.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    name: "Notify about changes made"
+    runs-on: ubuntu-latest
+    needs: [TestPods]
+    if: ${{ needs.TestPods.outputs.changed-files != '' &&  inputs.retake_snapshots}}
+    steps:
+      - name: Notify PR of screenshots changes
+        uses: actions/github-script@v6
+        with:
+          script: |
+            var body = `### Snapshots were updated. Please verify the changes match the expected layout. \n\n>'${{ needs.TestPods.outputs.changed-files }}'`;
```
Comment (marianeum) [inline .github/workflows/_test.yml:94]: Maybe include instructions here on how to record the snap shots? (and update the contributing guide to reflect the new flow as well)

**Pull request merged**